### ISSUE TITLE
Custom delimiter for CSV / SSV outputs

### ISSUE THREAD

**Issue by pedrooot** [toniblyx/prowler#4461]
### New feature motivation

Currently Prowler generates the CSV outputs using `;` as delimiter. The [CSV RFC](https://datatracker.ietf.org/doc/html/rfc4180) specifies that the delimiter should be `,`. The goal is to make a customizable delimiter for this output.

### Solution Proposed

Have a custom separator for the CSV output or create a new output type called `SSV` which has `;` as native separator and use `,` on the `CSV` output.

### Describe alternatives you've considered

If you have any trouble when opening a CSV file output from Prowler in LibreOffice or Excel we recommend you to follow these steps:

- LibreOffice Issue -> https://ask.libreoffice.org/t/change-column-separator-on-csv-export/8404/3
- Excel Issue -> https://www.ablebits.com/office-addins-blog/change-excel-csv-delimiter/

**Comment by abant07** [comment]
Hey @pedrooot 

Is this issue still open? If so, can I work on it?

**Comment by pedrooot** [comment]
Hey @abant07 I think that it's better if you start with #4703 
The fix for this issue should be evaluated by the team.
Thanks for helping!

**Comment by abant07** [comment]
Sounds good

**Comment by abant07** [comment]
Hey @pedrooot 

Am I okay to start on this issue?

**Comment by jfagoagas** [comment]
Hello @abant07 we are not sure about what to do here. If we decide that we want to add it we'll let you know.

Thanks!

**Comment by abant07** [comment]
Sounds good

**Comment by mitcharoo** [comment]
For anyone that finds this, a temporary solution I'm using is to replace commas (due to their presence in control descriptions/results/etc) with a space and then replace semicolons with commas:

`sed -e 's/,/  /g' -e 's/;/,/g'`

**Comment by toniblyx** [comment]
I think we can take advantage of v5 soon to have this sorted out. Leave default as it is (to not break anything) and add custom delimiter support option. Usage of `sed -e 's/,/  /g' -e 's/;/,/g'` is not recommended because some checks have comma in their fields, so that will break the CSV (or SSV) for sure.

**Comment by mitcharoo** [comment]
Yep, that's why I'm removing commas before replacing semicolons. As noted, a temporary workaround I've been using with no issue until Prowler supports the comma delimiter.

**Comment by Adithya1331** [comment]
Hi @pedrooot 👋

I'd love to contribute to this issue if it's available to take on. Could you please confirm if this is open for contributors to work on?

Also, could you clarify the expected output? Specifically:

* Should we retain `;` as the default delimiter for backward compatibility?
* Should `,` be the default for the CSV output, with a separate `--delimiter` flag to customize it?
* Is the idea to introduce a separate `SSV` output format that uses `;`, or simply make the delimiter configurable across formats?

**Comment by pedrooot** [comment]
Hey, @Adithya1331 the bad thing about this issue is that we're doing some changes in the metadata that could affect this issue. I think that It's better to wait until the metadata changes are applied to work on this one. Meanwhile, I'll get an answer for your questions:

* Yes, the `;` would be the default.
* I would say that the `;` should be the default one.
* The idea is to keep a single format (`CSV`) and change the delimiter.

Please, tell me what you think about this and we can discuss to reach the best approach.
